pkg/admission/cert/writer/certwriter.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package writer
+
+import (
+	"bytes"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"log"
+	"net/url"
+
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/admission/cert/generator"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// CACertName is the name of the CA certificate
+	CACertName = "ca-cert.pem"
+	// ServerKeyName is the name of the server private key
+	ServerKeyName = "key.pem"
+	// ServerCertName is the name of the serving certificate
+	ServerCertName = "cert.pem"
+)
+
+// CertWriter provides method to handle webhooks.
+type CertWriter interface {
+	// EnsureCert ensures that the webhooks have proper certificates.
+	EnsureCerts(runtime.Object) error
+}
+
+// Options are options for configuring a CertWriter.
+type Options struct {
+	Client        client.Client
+	CertGenerator generator.CertGenerator
+}
+
+// NewCertWriter builds a new CertWriter using the provided options.
+// By default, it builds a MultiCertWriter that is composed of a SecretCertWriter and a FSCertWriter.
```

Comment:
This document is not correct, it doesn't do the file writer

Reply:
This will happen in one week.
I can remove it for now.